src/daemon/context.rs

```diff
@@ -229,7 +230,7 @@ impl DaemonContext {
                     let payload = locator.to_sign();
                     if id_pk.fingerprint() == fingerprint {
                         id_pk.verify(&payload, &locator.signature)?;
-                        self.haven_dht.insert(fingerprint, locator.clone());
+                        self.local_rdht_shard.insert(fingerprint, locator.clone());
```

Comment:
this is what I mean, we should have a separate cache for lookups. The name makes it clear that this is not really what we want :)